internal/command/scale/count_machines.go

```diff
@@ -257,7 +257,7 @@ type planItem struct {
 	Region        string
 	Delta         int
 	Machines      []*api.Machine
-	MachineConfig *api.MachineConfig
+	MachineConfig api.MachineConfig
```

Comment:
why? it will make a shallow copy of the machine config, it doesn't prevent the copy from sharing other sub fields. It doesn't look like a sensible change IMO. If you need to clone the config before modifying it, use `helpers.Clone`

Reply:
Ah, nice, didn't know about `helper.Clone`, makes sense. The reason for the copy is due to this line:

https://github.com/superfly/flyctl/blob/f8080d10a19f095d6a2b08f7495cec20d3209378/internal/command/scale/count_machines.go#L166-L177

Since the same `action` is used if the delta is bigger than 1, it gets modified each iteration when we assign the new mount values. On the next iteration, `machineConfig.Mounts` has the values of the new volume. That's a problem mostly because we support restoring from a snapshot. The `GetVolumeSnapshots` takes the volume id. So if that changes to the newly created volume, it will have no snapshots and break the logic.